src/components/fields/field-array-source/field-array-source.component.tsx

```diff
@@ -74,41 +68,50 @@ export const FieldArraySource = (props): JSX.Element => {
         <div className="col-sm-7" />
       </div>
       {forholdTilKilde && forholdTilKilde !== 'egendefinert' && (
-        <div>
-          {getKilde(props).map((kilde, index) => {
-            return (
-              <div key={kilde.id} className="row d-flex fdk-after-element">
-                <div className="col-sm-5">
-                  <Field
-                    name={`kildebeskrivelse.kilde[${index}].tekst`}
-                    component={InputField}
-                    label={localization.titleSource}
-                    showLabel
-                  />
-                </div>
+        <FieldArray
+          name="kildebeskrivelse.kilde"
+          render={({ form, push }) => (
+            <div>
+              {getKilde(form).map((kilde, index) => {
+                return (
+                  <div key={`${kilde.id}-${index}`} className="row d-flex fdk-after-element">
+                    <div className="col-sm-5">
+                      <Field
+                        name={`kildebeskrivelse.kilde[${index}].tekst`}
+                        component={InputField}
+                        label={localization.titleSource}
+                        showLabel
+                      />
+                    </div>
 
-                <div className="col-sm-5">
-                  <Field
-                    className="fdk-after-element"
-                    name={`kildebeskrivelse.kilde[${index}].uri`}
-                    component={InputField}
-                    label={localization.linkSource}
-                    showLabel
-                  />
-                </div>
+                    <div className="col-sm-5">
+                      <Field
+                        className="fdk-after-element"
+                        name={`kildebeskrivelse.kilde[${index}].uri`}
+                        component={InputField}
+                        label={localization.linkSource}
+                        showLabel
+                      />
+                    </div>
 
-                <div className="fdk-source col-sm-2">
-                  <ButtonSource
-                    remove
-                    title={localization.removeSource}
-                    handleClick={() => handleRemoveKilde(props.form, index)}
-                  />
-                </div>
-              </div>
-            );
-          })}
-          <ButtonSource add title={localization.addNewSource} handleClick={() => handleAddKilde(props.form)} />
-        </div>
+                    <div className="fdk-source col-sm-2">
+                      <ButtonSource
+                        remove
+                        title={localization.removeSource}
+                        handleClick={() => handleRemoveKilde(form, index)}
+                      />
+                    </div>
+                  </div>
+                );
+              })}
+              <ButtonSource
+                add
```

Comment:
Litt for mye detaljer her i handleClick.
Kanskje det var bra å ha separat eksplisitt handleAddKilde. men da må du gi push som parameter til handler. 
Eller kanskje ha en constructor/factory til ny item. Har vi ikke noe domain layer til dette?
```
createSource=()=>{ id: v4(), tekst: '', uri: '' }
```

Reply:
ok